GVRf/Framework/src/org/gearvrf/GVRMonoscopicViewManager.java

```diff
@@ -0,0 +1,157 @@
+
+
+    /**
+     * Constructs GVRMonoscopicViewManager object with GVRScript which controls GL
+     * activities
+     * 
+     * @param gvrActivity
+     *            Current activity object
+     * @param gvrScript
+     *            {@link GVRScript} which describes
+     * @param distortionDataFileName
+     *            distortion filename under assets folder
+     */
+    GVRMonoscopicViewManager(GVRActivity gvrActivity, GVRScript gvrScript,
+            String distortionDataFileName) {
+        super(gvrActivity, gvrScript, distortionDataFileName);
+
+        /*
+         * Sets things with the numbers in the xml.
+         */
+        GVRXMLParser xmlParser = new GVRXMLParser(gvrActivity.getAssets(),
+                distortionDataFileName);
+        
+        mView = new GVRSurfaceView(gvrActivity, this, null);
+        gvrActivity.setContentView(mView);
+
+        DisplayMetrics metrics = new DisplayMetrics();
+        gvrActivity.getWindowManager().getDefaultDisplay().getMetrics(metrics);
+
+        final float INCH_TO_METERS = 0.0254f;
+        int screenWidthPixels = metrics.widthPixels;
+        int screenHeightPixels = metrics.heightPixels;
+        float screenWidthMeters = (float) screenWidthPixels / metrics.xdpi
+                * INCH_TO_METERS;
+        float screenHeightMeters = (float) screenHeightPixels / metrics.ydpi
+                * INCH_TO_METERS;
+
+        mLensInfo = new GVRLensInfo(screenWidthPixels, screenHeightPixels,
+                screenWidthMeters, screenHeightMeters, xmlParser);
+
+        GVRPerspectiveCamera.setDefaultFovY(xmlParser.getFovY());
+        int fboWidth = xmlParser.getFBOWidth();
+        int fboHeight = xmlParser.getFBOHeight();
+        if(fboWidth <= 0) {
+            fboWidth = screenWidthPixels;
+        }
+        if(fboHeight <=0) {
+            fboHeight = screenHeightPixels;
+        }
+        float aspect = (float) fboWidth / (float) fboHeight;
+        GVRPerspectiveCamera.setDefaultAspectRatio(aspect);
+        mViewportX = 0;
+        mViewportY = 0;
+        mViewportWidth = fboWidth;
+        mViewportHeight = fboHeight;
+        if(fboWidth != screenWidthPixels) {
+            mViewportX = (screenWidthPixels / 2) - (fboWidth /2);
+        }        
+        if(fboHeight != screenHeightPixels) {
+            mViewportY = (screenHeightPixels / 2) - (fboHeight / 2);
+        }
+        
+    }
+
+    /*
+     * GL life cycle
+     */
+    @Override
+    void onDrawFrame() {
+        //Log.v(TAG, "onDrawFrame");
+        mFrameHandler.beforeDrawEyes();
+        mFrameHandler.onDrawFrame();
+        mFrameHandler.afterDrawEyes();
+    }
+
+    @Override
+    protected void drawEyes() {
+        //Log.d(TAG, "drawEyes()");
+        mMainScene.getMainCameraRig().predict(3.5f / 60.0f);
+        GVRMonoscopicRenderer.renderCamera(mMainScene, mMainScene.getMainCameraRig().getLeftCamera(),
+                mViewportX, mViewportY, mViewportWidth, mViewportHeight, mRenderBundle);
```

Comment:
I like the general idea, but we don't appear to have a center camera in the CameraRig class.  We'd have to add one.  I'd be up for doing that in a different check-in, but this check-in already adds a ton of code.  If you don't mind, let's do that later.